app/src/main/java/com/example/wearit/components/MasterButton.kt

```diff
@@ -0,0 +1,75 @@
+package com.example.wearit.components
+
+import androidx.compose.foundation.border
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.*
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+
+
+enum class ButtonType(){
+    RED,
+    WHITE
+}
+
+@Composable
+fun MasterButton(
+    type: ButtonType,
+    onClick: () -> Unit,
+    modifier: Modifier,
+    icon: Int?,
+    text: String,
+    width: Dp,
```

Comment:
Ja bym żadnego argumentu width nie dawał, bo wtedy ten komponent jest dużo mniej elastyczny. Jeśli już naprawdę trzeba ustawić width na sztywno, to możesz to zrobić poprzez modifier (np. Modifier.width(180.dp)). Inna sprawa jest taka ze ja bym tego w ogóle nie ustawiał na sztywno, bo wtedy może się to wysypac na ekranie o innym rozmiarze. Jeśli chcesz mieć obok siebie dwa przyciski o równej szerokości, to możesz w obu ustawić modifier = Modifier.fillMaxWidth().weight(1f). Wtedy podzielą się równo miejscem miedzy sobą. A jak jeszcze chcesz dodać odstęp miedzy nimi to musisz to ustawić w Row'ie

Reply:
rzeczywiście lepiej jest to zrobić weightem, a odstęp ustaliłem paddingiem.